vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1199,9 +1199,6 @@ func (e *Store) calculateTTL(obj runtime.Object, defaultTTL int64, update bool)
 // present when the object is exported.
 func exportObjectMeta(accessor metav1.Object, exact bool) {
 	accessor.SetUID("")
-	if !exact {
-		accessor.SetNamespace("")
```

Comment:
@deads2k from what I can tell, it wasn't the export strategy that was defaulting an empty namespace to "default".
In the case of the imagestream I tested with, its `ExportStrategy` was nil. Instead, the `CreateStrategy` was [called here](https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L1231). I suspect it is somewhere there that the namespace is defaulted

Reply:
This code looks correct.  Check the return value from teh apiserver.  Was it actually "default" or was that set client-side?

Reply:
Taking a second look, this problem does not exist when testing against a cluster created with `openshift start`.

Can only replicate if using a cluster created via `oc cluster up`. Still looking

Reply:
@deads2k 
>  Was it actually "default" or was that set client-side?

It does not look like the client is setting it to "default", from what I can tell.
--loglevel=8 of `oc get is/jenkins -o yaml --export`: http://pastebin.test.redhat.com/553666

Reply:
@mfojtik @deads2k for comparison, here is `--loglevel 8` output for the same command, but executed against a cluster created via `openshift start` instead: http://pastebin.test.redhat.com/553669

Reply:
cc @bparees I'm not sure if `oc cluster up` uses a different export strategy / why it appears to behave differently than an `$ openshift start` cluster.

We handle server-side export here: https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L1203 clearing the namespace field for an object here https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L1203

Not sure how / where an empty namespace field could be defaulting to `default` as its value. The client does not appear to be altering this, as loglevel output linked to above shows that the cluster is already returning the object with the namespace field value defaulted.

This does not happen with `openshift start` clusters. I have tested with 3.8 and 3.9 releases of the openshift binary.

I have also tested with `oc cluster up --latest` and `oc cluster up --version v3.8.0-alpha.0`. Was hoping you had maybe seen something like this before?

Reply:
> Was hoping you had maybe seen something like this before?

i haven't, and no, oc cluster up certainly doesn't configure export strategies, so i can't imagine why there would be a difference in behavior.

also "oc export" seems to do the right thing (leaves off the namespace).

Anyway I'm seeing the "wrong" behavior using openshift start clusters, as well as oc cluster up clusters:

exporting from current project==openshift:

```
$ oc get --export is jenkins -o yaml
apiVersion: image.openshift.io/v1
kind: ImageStream
metadata:
  annotations:
    openshift.io/display-name: Jenkins
    openshift.io/image.dockerRepositoryCheck: 2018-02-07T19:19:34Z
  creationTimestamp: null
  generation: 1
  name: jenkins
  namespace: default
```